x/ecocredit/server/expiration.go

```diff
@@ -0,0 +1,51 @@
+package server
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/regen-network/regen-ledger/orm"
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+)
+
+// PruneOrders checks if there are any expired sell or buy orders and removes them from state.
+func (s serverImpl) PruneOrders(ctx sdk.Context) error {
+	blockTime := uint64(ctx.BlockTime().UnixNano())
+	minTime := uint64(0)
+
+	sellOrdersIter, err := s.sellOrderByExpirationIndex.PrefixScan(ctx, minTime, blockTime)
+	if err != nil {
+		return err
+	}
+
+	var sellOrders []*ecocredit.SellOrder
+	_, err = orm.ReadAll(sellOrdersIter, &sellOrders)
+	if err != nil {
+		return err
+	}
+
+	for _, order := range sellOrders {
+		err := s.sellOrderTable.Delete(ctx, order.OrderId)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
I think this is not efficient. We should have one loop where we read value and remove one by one